src/java_bytecode/generate_java_generic_method.h

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************\
+
+Module: Generate Java Generic Method - Instantiate a generic method
+        with concrete type information.
+
+Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#ifndef CPROVER_JAVA_BYTECODE_GENERATE_JAVA_GENERIC_METHOD_H
+#define CPROVER_JAVA_BYTECODE_GENERATE_JAVA_GENERIC_METHOD_H
+
+#include <util/message.h>
+#include <util/symbol_table.h>
+#include <util/std_types.h>
+#include <java_bytecode/java_types.h>
+
+class generate_java_generic_methodt
+{
+public:
+  typedef std::initializer_list
+    <std::pair<const symbol_typet, const symbol_typet>>
+      type_variable_instantiationst;
+
+  typedef std::unordered_map
+    <const symbol_typet, const symbol_typet, irep_hash>
+      type_variable_instantiation_mapt;
+
+  explicit generate_java_generic_methodt(message_handlert &message_handler);
+
+  const symbolt& operator()(
+    const symbolt &generic_method,
+    const type_variable_instantiationst &concrete_types,
+    symbol_tablet &symbol_table) const;
+
+  const std::string instantiation_decoration(
+    type_variable_instantiationst concrete_types) const;
+
+  const void instantiate_java_generic_parameter(
+    java_generic_parametert &generic_parameter,
+    const type_variable_instantiation_mapt &concrete_types) const;
+
+  const void instantiate_java_generic_type(
+    java_generic_typet &generic_type,
+    const type_variable_instantiation_mapt &concrete_types) const;
+
+  void decorate_identifier(
+    irept &expr,
+    const dstringt &identifier,
+    const dstringt &pattern,
+    const dstringt &decoration) const;
+
+private:
+  message_handlert &message_handler;
```

Comment:
I don't think we use this anywhere so can probably just be removed and then I suppose this can just become a free function (personally prefer it like this as it allows you to forward declare the methods in the header without having to expose them).